### PR TITLE
Add new node into the replica which has the most of offline nodes

### PR DESCRIPTION
**Pull request by yah01**
Signed-off-by: yah01 <yang.cen@zilliz.com>
#16705 
This PR won't totally solve the issue, but it can handle some cases, the issue will be solved after implementing the balancing between replicas.

### CODE REVIEW

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

File: internal/querycoord/group_balance.go [hunk 1]
```diff
@@ -34,8 +35,17 @@ func (b *replicaBalancer) addNode(nodeID int64) ([]*balancePlan, error) {
...
+			for _, nodeID := range replica.NodeIds {
+				if isOnline, err := b.cluster.isOnline(nodeID); err != nil || !isOnline {
+					offlineNodesCnt[replica.ReplicaID]++
+				}
+			}
+		}
 		sort.Slice(replicas, func(i, j int) bool {
-			return len(replicas[i].GetNodeIds()) < len(replicas[j].GetNodeIds())
+			return offlineNodesCnt[replicas[i].ReplicaID] > offlineNodesCnt[replicas[j].ReplicaID]
 		})
```
Comment (congqixia) [inline internal/querycoord/group_balance.go:49]: If all replica nodes are online(offlineCnt=0), the new node will be added randomly. Can we add it to the group with least available memory?

Comment (yah01) [reply internal/querycoord/group_balance.go]: OK, now this sorts them with the number of offline nodes as first key and the available memory as the second key

**Comment by yah01** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**